pkg/agent/cache/manager/manager.go

```diff
@@ -0,0 +1,302 @@
+package manager
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+
+	"crypto/tls"
+	"crypto/x509"
+	"github.com/sirupsen/logrus"
+	spiffe_tls "github.com/spiffe/go-spiffe/tls"
+	"github.com/spiffe/go-spiffe/uri"
+	"github.com/spiffe/spire/pkg/agent/cache"
+	"github.com/spiffe/spire/pkg/common/util"
+	"github.com/spiffe/spire/proto/api/node"
+	proto "github.com/spiffe/spire/proto/common"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"io"
+	"sync"
+	"time"
+)
+
+type Manager interface {
+	UpdateCache(stop chan struct{})
+	fetchSVID(requests []EntryRequest, nodeClient node.NodeClient, wg *sync.WaitGroup)
+	getGRPCConn(svid []byte, key *ecdsa.PrivateKey) (*grpc.ClientConn, error)
+	expiredCacheEntryHandler(stop chan struct{})
```

Comment:
Interface with unexported methods?

Reply:
this interface will be used to inject mocks in tests